app/services/anonymous_shadow_creator.rb

```diff
@@ -55,6 +55,14 @@ def create_shadow!
           trust_level: 1,
           manual_locked_trust_level: 1,
           approved: true,
+          groups:
+            (
+              if SiteSetting.allow_anonymous_to_inherit_group
```

Comment:
This SiteSetting might be redundant, since groups default to be false and admins must opt groups into inheritance.  

I'm leaning towards leaning on the side of more protections for user anonymity, so I'll keep it in there fore now.